### PR TITLE
[5.9] Support Target Grouping

### PR DESCRIPTION
**Pull request by elsh**
Support Target Grouping
- Allow opting out of package boundary
- Default set to package boundary

Resolves rdar://107454728

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test windows

**Review by tomerd — APPROVED** [review]

**Review by nkcsgexi — APPROVED** [review]

File: Sources/CompilerPluginSupport/TargetExtensions.swift [hunk 1]
```diff
@@ -16,12 +16,14 @@ public extension Target {
     @available(_PackageDescription, introduced: 999.0)
     static func macro(
         name: String,
+        group: TargetGroup,
```
[the remark this answers is not in the document]
Comment (elsh) [reply Sources/CompilerPluginSupport/TargetExtensions.swift:19]: https://github.com/apple/swift-package-manager/pull/6385

**Pull request merged**